### PR TITLE
refactor(workspace): convert the project in a workspace

### PR DESCRIPTION
**Pull request by joshuachp**
This now depends on #241

### CODE REVIEW

**Comment by joshuachp** [comment]
Still a draft, I want to see if the CI whould run correctly

File: .cargo/config.toml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# This file is part of Hedgehog.
```
Comment (harlem88) [inline .cargo/config.toml:1]:
```suggestion
# This file is part of Edgehog.
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,11 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## 0.7.2 - Unreleased
```
Comment (harlem88) [inline CHANGELOG.md:7]:
```suggestion
## [Unreleased]
```

File: Cargo.toml [hunk 3]
```diff
@@ -3,61 +3,54 @@
...
 version = "0.7.1"
 default-run = "edgehog-device-runtime"
 edition = "2021"
 homepage = "https://github.com/edgehog-device-manager/edgehog-device-runtime"
-rust-version = "1.59.0"
+rust-version = "1.66.1"
 
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
```
Comment (harlem88) [inline Cargo.toml:24]: Specifying a workspace dependency to be inherited by members of a workspace.
```suggestion
[workspace.dependencies]
```

File: e2e-test/Cargo.toml [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+[package]
+name = "e2e-test"
+version = "0.1.0"
+edition = "2021"
+rust-version = "1.59.0"
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+
+[dependencies]
+astarte-device-sdk = { version = "0.5.1", features = ["derive"] }
```
Comment (harlem88) [inline e2e-test/Cargo.toml:28]:
```suggestion
astarte-device-sdk = { workspace = true, features = ["derive"] }
```

**Review by harlem88 — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 5]
```diff
@@ -4,6 +4,11 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [Unreleased]
+## Changed
+
```
Comment (harlem88) [inline CHANGELOG.md:9]:
```suggestion

```

**Review by harlem88 — CHANGES_REQUESTED** [review]

**Review by harlem88 — APPROVED** [review]

**Pull request merged**